reference/src/client/test/sa_key_provision_applemfi.cpp

```diff
@@ -0,0 +1,404 @@
+        //into sa_key_provision_ta, how to use provision object data is all up to
+        //SOC vendors.
+        appleMfiProvision->mfiProvisioningObject = (void*)pb_key_container.data();
+        appleMfiProvision->mfiProvisioningObjectLength = pb_key_container.size();
+
+        status = sa_key_provision_ta(APPLE_MFI_PROVISIONING, appleMfiProvision,
+           sizeof(AppleMfiProvisioning), nullptr);
+        ASSERT_EQ(status, SA_STATUS_OK);
+
+        if (nullptr != appleMfiProvision) {
+           delete appleMfiProvision;
+           appleMfiProvision = nullptr;
+        }
+    }
+    TEST_F(SaKeyProvisionAppleMfiTest, failsZeroInLength) {
+        auto key_length = SYM_128_KEY_SIZE;
+        std::vector<uint8_t> clear_base_key = random(key_length);
+
+        //create a base key container
+        auto base_key_type = SA_KEY_TYPE_SYMMETRIC;
+        std::string base_key_type_string = "AES-128";
+        std::string base_key_container;
+        sa_status status = create_key_container(
+             base_key_type_string, //key type string
+             base_key_type, //key type
+             clear_base_key,
+             base_key_container);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        INFO("base_key_container length : %d", base_key_container.size());
+        //Note: here just provide an examlpe to create base key and passed
+        //into sa_key_provision_ta, how to use base key is all up to
+        //SOC vendors.
+        AppleMfiProvisioning *appleMfiProvision = new AppleMfiProvisioning;
+        ASSERT_NE(nullptr, appleMfiProvision);
+        appleMfiProvision->mfiBaseKey = (void*)base_key_container.data();
+        appleMfiProvision->mfiBaseKeyLength = base_key_container.size();
+
+
+        //create a provisioning object container
+        auto pb_key_type = SA_KEY_TYPE_SYMMETRIC;
+        std::string pb_key_type_string = "AES-128";
+        std::vector<uint8_t> clear_provision_object = random(key_length);
+        std::string pb_key_container;
+        status = create_key_container(
+             pb_key_type_string, //key type string
+             pb_key_type, //key type
+             clear_provision_object,
+             pb_key_container);
+        ASSERT_EQ(SA_STATUS_OK, status);
+        INFO("pb_key_container length : %d", pb_key_container.size());
+        //Note: here just provide an examlpe to create provision object and passed
+        //into sa_key_provision_ta, how to use provision object data is all up to
+        //SOC vendors.
+        appleMfiProvision->mfiProvisioningObject = (void*)pb_key_container.data();
+        appleMfiProvision->mfiProvisioningObjectLength = pb_key_container.size();
+
+        status = sa_key_provision_ta(APPLE_MFI_PROVISIONING, appleMfiProvision,
+           0, nullptr);
+
+        if (nullptr != appleMfiProvision) {
+           delete appleMfiProvision;
+           appleMfiProvision = nullptr;
+        }
+        ASSERT_NE(status, SA_STATUS_OK);
```

Comment:
We should use a similar test strategy as other SecAPI3 tests where the failure that is being tested is asserted. Looks like this applies to all tests in this PR. Use other secAPI3 client tests as a guide such as those for sa_key_import.